apps/storybook/src/stories/transfer-button/transfer-button.stories.tsx

```diff
@@ -11,16 +10,28 @@ interface TransferButtonProps {
 }
 
 const TransferButtonStory = ({ to, account }: TransferButtonProps) => {
-  const [amount, setAmount] = useState(0.001);
+  const [amount, setAmount] = useState("0.001");
 
   return (
     <div className="flex flex-col gap-2">
       <Label>Amount</Label>
-      <div className="flex gap-2 relative">
-        <Input type="number" value={amount} onChange={(e) => setAmount(Number(e.target.value))} />
-        <p className="absolute right-3 top-0 bottom-0 flex items-center">ETH</p>
+      <div className="flex items-center gap-2 relative border border-primary rounded-md pr-2 overflow-hidden">
+        <input
+          placeholder="0"
+          pattern="[1-9.]*"
+          value={String(amount)}
+          onChange={(e) => {
+            const value = e.target.value;
+            // Only allow numbers and a single decimal point
+            if (/^[0-9]*\.?[0-9]*$/.test(value)) {
```

Comment:
This regex also matches empty string because everything is optional.

```suggestion
            if (/^([0-9]+\.)?[0-9]+$/.test(value)) {
```

Reply:
Would be best if you can throw in a test case for this. Not required to merge though.